File: implementations/react/src/patterns/behavioral/command/invoker.ts
import type { Command } from "./command-interface";

/**
 * Invocador en el patrón Command.
 */


export class Invoker {
    private command : Command;

    constructor(command: Command){
        this.command = command;
    }

    /** Establece el comando a ejecutar */
    setCommand(command: Command){
        this.command = command;
    }

    /** Ejecuta el comando establecido */
    executeCommand(item?: string){
        this.command.execute(item);
    }
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/behavioral/command/invoker.dart
import 'package:design_patterns_flutter/patterns/behavioral/command/command.dart';

class Invoker {
  Command command;
  Invoker(this.command);

  void setCommand(Command c) {
    command = c;
  }

  void executeCommand([String? item]) {
    command.execute(item);
  }
}
